app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -260,6 +260,15 @@ def add_middleware_deployment(ems_ref, hash)
           :binary_content        => hash[:file]["file"].read,
           :resource_path         => ems_ref.to_s
         }
+        unless hash[:file]['server_groups'].nil?
+          # in case of deploying into server group the resource path should point to the domain controller
+          deployment_data[:server_groups] = hash[:file]['server_groups']
+          server_group_path = ::Hawkular::Inventory::CanonicalPath.parse(deployment_data[:resource_path])
+          server_group_path_hash = server_group_path.to_h
+          server_group_path_hash[:resource_ids] = [server_group_path_hash[:resource_ids].first]
```

Comment:
@Jiri-Kremser - maybe an alternative, slightly simpler, way to do the same thing:

 ```
server_group_path_hash = ::Hawkular::Inventory::CanonicalPath.parse(deployment_data[:resource_path]).to_h
server_group_path_hash[:resource_ids].slice!(1..-1)
 ```